cli/util_internal_test.go

```diff
@@ -41,6 +41,37 @@ func TestDurationDisplay(t *testing.T) {
 	}
 }
 
+func TestExtendedParseDuration(t *testing.T) {
+	t.Parallel()
+	for _, testCase := range []struct {
+		Duration   string
+		Expected   time.Duration
+		ExpectedOk bool
+	}{
+		{"1d", 24 * time.Hour, true},
+		{"1y", 365 * 24 * time.Hour, true},
+		{"10s", 10 * time.Second, true},
+		{"1m", 1 * time.Minute, true},
+		{"20h", 20 * time.Hour, true},
+		{"10y10d10s", 10*365*24*time.Hour + 10*24*time.Hour + 10*time.Second, true},
+		{"10ms", 10 * time.Millisecond, true},
+		{"5y10d10s5y2ms8ms", 10*365*24*time.Hour + 10*24*time.Hour + 10*time.Second + 10*time.Millisecond, true},
+		{"10yz10d10s", 0, false},
```

Comment:
question: What happens if we try to parse a duration greater than `2540400h10m10.000000000s` (e.g. `1000y`)?
(see: https://github.com/golang/go/blob/master/src/time/time.go#L954C21-L954C45)